integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/receive/PayloadConverter.java

```diff
@@ -11,10 +11,24 @@
 
     private final JAXBContext jaxbContext;
     private Class<T> clazz;
+    private String namespaceUri;
+    private String localPart;
 
-    public PayloadConverter(Class<T> clazz) {
+    public PayloadConverter(Class<T> clazz, String namespaceUri, String localPart) {
         this.clazz = clazz;
+        this.namespaceUri = namespaceUri;
+        this.localPart = localPart;
+        try {
+            jaxbContext = JAXBContext.newInstance(clazz);
+        } catch (JAXBException e) {
+            throw new RuntimeException("Could not create JAXBContext for " + clazz);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00112)